File: Toll-Calculator/Models/Vehicles/Military.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toll_Calculator.Enums;
using Toll_Calculator.Interfaces;

namespace Toll_Calculator.Models.Vehicles
{
    public class Military : IVehicle
    {
        public bool IsTollFree()
        {
            return true;
        }

        public VehicleType GetVehicleType()
        {
            return VehicleType.Military;
        }
    }
}

File: Golang/vehicle/military.go
package vehicle

// military struct for handling Military type.
type military struct{}

// GetVehicleType will return the "Military" type.
func (m military) GetVehicleType() string {
	return "Military"
}

// IsTollFree return true for Military type.
func (m military) IsTollFree() bool {
	return true
}

// NewMilitary return a vehicle type of Military.
func NewMilitary() Vehicle {
	return &military{}
}
